File: python/jwt/security/audit/jwt-exposed-data.py
import jwt

def bad1(secret, payload):
    # ruleid: jwt-python-exposed-data
    encoded = jwt.encode(payload, secret, algorithm='HS256')
    return encoded

def ok(secret_key):
    # ok: jwt-python-exposed-data
    encoded = jwt.encode({'some': 'payload'}, secret_key, algorithm='HS256')
    return encoded

File: ruby/jwt/security/audit/jwt-exposed-data.rb
require 'jwt'

def bad1(hmac_secret, payload)
    # ruleid: ruby-jwt-exposed-data
    token = JWT.encode payload, hmac_secret, 'HS256'
    puts token
end

def ok1(hmac_secret)
    # ok: ruby-jwt-exposed-data
    payload = { data: 'data', nbf: nbf }
    token = JWT.encode payload, hmac_secret, 'HS256'
    puts token
end
